client-lib/src/main/java/com/ibm/ws/repository/transport/client/LooseFileClient.java

```diff
@@ -69,56 +71,54 @@ public void checkRepositoryStatus() throws IOException {
         }
     }
 
-    /** {@inheritDoc} */
-    @Override
-    public Asset getAsset(final String assetId) throws IOException, BadVersionException, RequestFailureException {
-        return getAsset(assetId, false);
-    }
-
-    /*
-     * ------------------------------------------------------------------------------------------------------------------
-     * PROTECTED AND OVERRIDEABLE IMPLEMENTATION METHODS
-     * ------------------------------------------------------------------------------------------------------------------
+    /**
+     * {@inheritDoc}
      */
-
     @Override
-    protected boolean exists(final String relative) {
-        return DirectoryUtils.exists(new File(relative));
+    public Collection<Asset> getAllAssets() throws IOException, RequestFailureException {
+        Collection<Asset> allAssets = new ArrayList<Asset>();
+        for (File f : assets) {
+            try {
+                allAssets.add(getAsset(f.getAbsolutePath()));
+            } catch (BadVersionException e) {
+                // Ignore assets with unknown versions
+            }
+        }
+        return allAssets;
     }
 
+    /** {@inheritDoc} */
     @Override
-    /**
-     * We treat all files as being under the "root" (i.e. when relative is the empty string)
-     */
-    protected boolean hasChildren(final String relative) {
-        return relative.equals("");
+    public Asset getAsset(final String assetId) throws IOException, BadVersionException, RequestFailureException {
```

Comment:
Maybe split this into two methods: `getAsset(String assetId)` which calls `getAsset(File assetFile)`

Then Line 82 can just call the `File` version